rad/prelude/test-eval.rad

```diff
@@ -0,0 +1,58 @@
+{:module 'prelude/test-eval
+ :doc
+   "Provides eval that adds a `:test` macro. If an the eval encounters an
+   expression of the form `(:test \"desc\" tests...)` we add it to the vector
+   stored in the `tests` ref. The tests are then run by `rad/prelude/test.rad`."
+ :exports '[eval tests]}
+
+(def tests
+  "Reference that collects tests defined with the `:test` macro. This
+  value is updated by a custom `eval`."
+  (ref []))
+
+;; Annoyingly verbose since we have neither the prelude nor pattern-matching.
+(def eval
+  "Evaluation function that adds :test macro to register tests."
```

Comment:
The annoying thing here is in the repl (and the reference doc), the doc for `eval` now mainly talks about tests.

Reply:
I’ll fix this in an upcoming PR.